Source/inv.cpp

```diff
@@ -704,6 +708,65 @@ bool AutoPlaceItemInInventory(int playerNumber, const ItemStruct &item, bool per
 	return done;
 }
 
+/**
+ * @brief Checks whether the given item can be placed on the specified player's inventory.
+ * If 'persistItem' is 'True', the item is also placed in the inventory, using an alternative
+ * algorithm that should typically provide better packing than AutoPlaceItemInInventory().
+ * @param playerNumber The player number on whose inventory will be checked.
+ * @param item The item to be checked.
+ * @param persistItem Pass 'True' to actually place the item in the inventory. The default is 'False'.
+ * @return 'True' in case the item can be placed on the player's inventory and 'False' otherwise.
+ */
+bool AutoPlaceItemInInventoryAlternate(int playerNumber, const ItemStruct &item, bool persistItem)
+{
+	InvXY itemSize = GetInventorySize(item);
+	bool done = false;
+
+	if (itemSize.Y == 1) {
+		for (int i = 30; i <= 39 && !done; i++) {
+			done = AutoPlaceItemInInventorySlot(playerNumber, i, item, persistItem);
+		}
+		for (int x = 9; x >= 0 && !done; x--) {
+			for (int y = 2; y >= 0 && !done; y--) {
+				done = AutoPlaceItemInInventorySlot(playerNumber, 10 * y + x, item, persistItem);
+			}
+		}
+	}
+
+	if (itemSize.Y == 2) {
+		for (int x = 10 - itemSize.X; x >= 0 && !done; x -= itemSize.Y) {
+			for (int y = 0; y < 3 && !done; y++) {
+				done = AutoPlaceItemInInventorySlot(playerNumber, 10 * y + x, item, persistItem);
+			}
+		}
+		if (itemSize.X == 2) {
+			for (int x = 7; x >= 0 && !done; x -= 2) {
+				for (int y = 0; y < 3 && !done; y++) {
+					done = AutoPlaceItemInInventorySlot(playerNumber, 10 * y + x, item, persistItem);
+				}
+			}
+		}

```

Comment:
this doesn't seam useful, all x2 items should be served fine by y3 or y2 paths

Reply:
This is the counterpart to the `x -= itemSize.X` issue, which is that this placement first attempts to align 2x2 objects by placing them in even columns, and only if it didn't succeed will it then try again using odd-column alignment, to prevent the `xoox` problem you mentioned above.